sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileSourceCheckpointData.cs

```diff
@@ -2,15 +2,48 @@
 // Licensed under the MIT License.
 
 using System.IO;
+using Azure.Core;
 
 namespace Azure.Storage.DataMovement.Files.Shares
 {
     internal class ShareFileSourceCheckpointData : StorageResourceCheckpointData
     {
+        /// <summary>
+        /// Schema version.
+        /// </summary>
+        public int Version;
+
         public override int Length => 0;
 
+        public ShareFileSourceCheckpointData()
```

Comment:
My thought with a checkpoint data class that doesn't contain anything is that we leave it blank as it seems unnecessary to just serialize a version?

Unless we plan to add something here in the near future (not sure if we will need it or not).

Reply:
Are you suggesting we serialize/deserialize zero bytes or that this class gets removed altogether?

Reply:
I'm suggesting the former. We leave this class as is with the length = 0 and the serialize method a no-op. We leave the class though, so we have an object to return in `GetSourceCheckpointData`. This is how I did it for the local checkpoint data that is currently empty.

I did have a thought to make one shared "empty" implementation that we could use in any cases where we don't yet have a specific implementation but not sure.